build/MinimalThirdPartyAllwmake

```diff
@@ -0,0 +1,244 @@
+    && make $configOpt install
+
+    # cleanup, could also remove Makefile.inc
+    make realclean 2>/dev/null
+)
+fi
+
+# verify existence of scotch include
+[ -f $SCOTCH_ARCH_PATH/include/scotch.h ] || {
+    echo
+    echo "    WARNING: required include file 'scotch.h' not found!"
+    echo
+}
+
+
+# build ptscotch if MPI (ThirdParty or system) is available
+if [ "${FOAM_MPI:-dummy}" != dummy ]
+then
+    echo ========================================
+    echo "Build PTScotch decomposition library $SCOTCH_VERSION (uses MPI)"
+    echo "    $SCOTCH_ARCH_PATH"
+    echo
+
+    if [ -f $SCOTCH_ARCH_PATH/include/$FOAM_MPI/ptscotch.h \
+      -a -r $FOAM_EXT_LIBBIN/$FOAM_MPI/libptscotch.so \
+      -a -r $FOAM_EXT_LIBBIN/$FOAM_MPI/libptscotcherrexit.so ]
+    then
+        echo "    ptscotch header in $SCOTCH_ARCH_PATH/include/$FOAM_MPI"
+        echo "    ptscotch libs   in $FOAM_EXT_LIBBIN/$FOAM_MPI"
+        echo
+    else
+    (
+        set -x
+        cd $SCOTCH_VERSION/src || exit 1
+
+        prefixDIR=$SCOTCH_ARCH_PATH
+        libDIR=$FOAM_EXT_LIBBIN/$FOAM_MPI
+        incDIR=$SCOTCH_ARCH_PATH/include/$FOAM_MPI
+
+        mkdir -p $prefixDIR 2>/dev/null
+        mkdir -p $libDIR 2>/dev/null
+
+        configOpt="prefix=$prefixDIR libdir=$libDIR includedir=$incDIR"
+
+        if [ -f $scotchMakefile ]
+        then
+            rm -f Makefile.inc
+            ln -s $scotchMakefile Makefile.inc
+        fi
+
+        [ -f Makefile.inc ] || {
+            echo " Error: scotch needs an appropriate Makefile.inc"
+            exit 1
+        }
+
+        # handle non-gcc compilers
+        unset configEnv
+        [ "${WM_CC:-gcc}" != gcc ] && configEnv="CC=$WM_CC CCS=$WM_CC"
+
+        make realclean 2>/dev/null  # for safety
+
+        make -j $WM_NCOMPPROCS $configEnv ptscotch \
+        && make $configOpt install
+
+        # cleanup, could also remove Makefile.inc
+        make realclean 2>/dev/null
+    )
+    fi
+
+    # verify existence of scotch include
+    [ -f $SCOTCH_ARCH_PATH/include/$FOAM_MPI/ptscotch.h ] || {
+        echo
+        echo "    WARNING: required include file 'ptscotch.h' not found!"
+        echo
+    }
+fi
+
+# Zoltan is optional
+echo ========================================
+echo Build Zoltan decomposition
+
+# Get ZOLTAN_VERSION, ZOLTAN_ARCH_PATH
+if settings=`$WM_PROJECT_DIR/bin/foamEtcFile config.sh/zoltan`
+then
+    . $settings
+fi
+if [ -d "$ZOLTAN_VERSION" ]
+then
+
+    if [ -f $ZOLTAN_ARCH_PATH/include/zoltan.h \
+      -a -r $FOAM_EXT_LIBBIN/$FOAM_MPI/libzoltan.a ]
+    then
+        echo "    Zoltan header in $ZOLTAN_ARCH_PATH/include"
+        echo "    Zoltan libs   in $FOAM_EXT_LIBBIN/$FOAM_MPI"
+        echo
+    else
+    (
+        set -x
+        cd $ZOLTAN_VERSION  || exit 1
+        mkdir -p build
+        cd build
+        ../configure \
+            --prefix=$ZOLTAN_ARCH_PATH \
+            --libdir=$FOAM_EXT_LIBBIN/$FOAM_MPI \
+            --with-mpi=/usr/local/mpi \
```

Comment:
@colemanjs we need to figure out why i needed to manually specify these. when you get a chance could you pull this repo and test if you can get it to work with the original `$MPI_ARCH_PATH` variable? if so which module(s) did you need?